### ISSUE TITLE
TypeScript files lack syntax highlighing

### ISSUE THREAD

**Issue by thomcc** [devtools-html/debugger.html#1642]
I have a typescript project, in the debugger, none of the TypeScript sources (`*.ts` files) are highlighted. This is unfortunate.

I suspect even just using JS for the default would work for the vast majority compiled to JS source files. I also suspect that a few (CoffeeScript, TypeScript, etc) are common enough that special casing them would reasonable (unless it would require shipping tons of additional syntax highlighting file definitions or something).

**Comment by clarkbw** [comment]
This is likely about supporting [CodeMirror TypeScript mode](https://codemirror.net/mode/javascript/typescript.html) which is just a one off of [CodeMirror JS mode](https://codemirror.net/mode/javascript/index.html).  I'm not sure if we get the right MIME typing to be able to differentiate TypeScript or not.

**Comment by jasonLaster** [comment]
Lets create a debugger example [here](https://devtools-html.github.io/debugger-examples/)

We can set the mode here:

```patch
diff --git a/src/components/Editor.js b/src/components/Editor.js
index 0aaafd1..f7c8306 100644
--- a/src/components/Editor.js
+++ b/src/components/Editor.js
@@ -302,6 +302,8 @@ const Editor = React.createClass({
   setMode(sourceText) {
     const contentType = sourceText.get("contentType");

+    // => we can set the mode here
+
     if (contentType.includes("javascript")) {
       this.editor.setMode({ name: "javascript" });
     } else if (contentType === "text/wasm") {
```

**Comment by jbhoosreddy** [comment]
This looks interesting. How about I take a look at this one next @jasonLaster?

**Comment by jbhoosreddy** [comment]
okay. some initial questions on how to proceed. @jasonLaster @clarkbw 

1. my local installation has the latest version of codemirror@5.22.0 but it is missing the code to support typescript. [javascript/typescript](https://github.com/codemirror/CodeMirror/blob/master/mode/javascript/typescript.html)
Any suggestions on dealing with it in a nice way?
2. I am looking at a small set of languages which would be nice to support. Coffeescript, Typescript and JSX come to mind. Any thoughts on this?

**Comment by jasonLaster** [comment]
Nice. Elm.

Best thing to do is to add examples to the debugger examples project as
well.

We can bump codemirror if it helps.

**Comment by jasonLaster** [comment]
@thomcc making some progress in #1687

**Comment by jbhoosreddy** [comment]
fixed in #1687
